### DISCUSSION TITLE
Linux packaging

### DISCUSSION THREAD

**Discussion by JamesWilmot** [thibauult/open-hue#31] [Ideas]
Hi @thibauult!

Thanks for putting together this excellent project. I don't own any Philips Hue devices but I have a project which I think I'll use them for, especially if I can control them through command line.

Do you have any interest in packaging the project for linux distros? I do want to have a crack at packaging something for NixOS so I can volunteer to do that.

**Comment by thibauult** [comment]
Hi @JamesWilmot, 

I am deeply sorry for missing this discussion here! 

As a Mac user, I focused on providing the Mac-related packages first (+ the Docker Image). I totally agree that it'd make sense to provide packages for Linux as well.

I had a quick look at the [GoReleaser options](https://goreleaser.com/customization/nfpm) and it seems pretty easy to generate the Linux packages. 

As I won't be able to test it on my side, I'd be glad if you could contribute!

**Comment by JamesWilmot** [comment]
Nah, all good. I would assume (and hope!) we all have lives outside of work, projects etc. 

I wouldn't mind having a crack at packaging for Ubuntu, Debian and NixOS. Useful learning experience for me as I might have to do something similar for work soon. 

I'll have a bit of a ponder about how to best progress this and then maybe create a handful of issues.

**Reply by thibauult** [reply]
Thanks @JamesWilmot, please let me known how I can help you. 

In the meantime, you can also note that some Linux (static) binaries are uploaded along with the [GitHub releases](https://github.com/openhue/openhue-cli/releases/tag/0.15): 

![SCR-20240502-ipoo](https://github.com/openhue/openhue-api/assets/39826516/83713669-a8dd-4e94-be30-f640f5f2ee64)

It's perhaps something you could start using as of now?
